packages/Discretization/test/tstPointSearch.cpp

```diff
@@ -0,0 +1,457 @@
+    Kokkos::View<DTK_CellTopology *, DeviceType> cell_topologies_view;
+    Kokkos::View<unsigned int *, DeviceType> cells;
+    Kokkos::View<double **, DeviceType> coordinates;
+    std::vector<unsigned int> n_subdivisions = {{5, 5, 3}};
+    std::tie( cell_topologies_view, cells, coordinates ) =
+        buildStructuredMesh<DeviceType>( comm, n_subdivisions );
+
+    Kokkos::View<double * [dim], DeviceType> points_coord( "points_coord", 1 );
+    auto points_coord_host = Kokkos::create_mirror_view( points_coord );
+    for ( unsigned int i = 0; i < dim; ++i )
+        points_coord_host( 0, i ) = 10000.;
+    Kokkos::deep_copy( points_coord, points_coord_host );
+
+    // We are now done with building the mesh and we can do the search
+
+    DataTransferKit::PointSearch<DeviceType> pt_search(
+        comm, cell_topologies_view, cells, coordinates, points_coord );
+
+    Kokkos::View<int *, DeviceType> ranks;
+    Kokkos::View<int *, DeviceType> cell_indices;
+    Kokkos::View<DataTransferKit::Point *, DeviceType> reference_points;
+    Kokkos::View<unsigned int *, DeviceType> query_ids;
+    std::tie( ranks, cell_indices, reference_points, query_ids ) =
+        pt_search.getSearchResults();
+
+    TEST_EQUALITY( ranks.extent( 0 ), 0 );
+    TEST_EQUALITY( cell_indices.extent( 0 ), 0 );
+    TEST_EQUALITY( reference_points.extent( 0 ), 0 );
+    TEST_EQUALITY( query_ids.extent( 0 ), 0 );
+}
+
+TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( PointSearch, two_topo_two_dim, DeviceType )
+{
+    // Test a mesh of made of Quadrilateral<4> and Triangle<3>
+    // 7-----8-----9
+    // |    / \    |
+    // 3---4---5---6
+    // |    \ /    |
+    // 0-----1-----2
+    //
+    // The pattern above is repeated on each processors
+
+    Teuchos::RCP<const Teuchos::Comm<int>> comm =
+        Teuchos::DefaultComm<int>::getComm();
+    int const comm_rank = comm->getRank();
+    int const comm_size = comm->getSize();
+    unsigned int constexpr dim = 2;
+    unsigned int const ref_rank = ( comm_rank + 1 ) % comm_size;
+    Kokkos::View<DTK_CellTopology *, DeviceType> cell_topologies_view;
+    Kokkos::View<unsigned int *, DeviceType> cells;
+    Kokkos::View<double **, DeviceType> coordinates;
+    Kokkos::View<double **, DeviceType> points_coord;
+
+    std::tie( cell_topologies_view, cells, coordinates ) =
+        buildMixedMesh<DeviceType>( comm, 2 );
+    points_coord = getPointsCoord2D<DeviceType>( comm );
+
+    // We are now done with building the mesh and we can do the search
+    DataTransferKit::PointSearch<DeviceType> pt_search(
+        comm, cell_topologies_view, cells, coordinates, points_coord );
+
+    Kokkos::View<int *, DeviceType> ranks;
+    Kokkos::View<int *, DeviceType> cell_indices;
+    Kokkos::View<DataTransferKit::Point *, DeviceType> reference_points;
+    Kokkos::View<unsigned int *, DeviceType> query_ids;
+    std::tie( ranks, cell_indices, reference_points, query_ids ) =
+        pt_search.getSearchResults();
+
```

Comment:
Based on the usage in these unit tests it is still not clear to me why this entire class cannot be static such that:

```
std::tie( ranks, cell_indices, reference_points, query_ids ) =
    DataTransferKit::PointSearch<DeviceType>::search(
        comm, cell_topologies_view, cells, coordinates, points_coord );
```

would be the way to use it. Does it need state if `getSearchResults()` is the only real public function a user will use? They can then easily cache the search results if they want to reuse them.

Reply:
Because I want the private data for the interpolation operator.  Your code would work but then the first thing I will need to do for the interpolation is a MPI communication. `getSearchResults()` sends the data back to the calling processors but I need to do more work when I do the interpolation.

Reply:
So you are going to add additional accessor functions so you can access that private data at a later time?

Reply:
No I am just going to make it `friend`